crates/node/src/kitchensink.rs

```diff
@@ -0,0 +1,506 @@
+            .arg("Unsafe")
+            .env(
+                "RUST_LOG",
+                "error,evm=debug,sc_rpc_server=info,runtime::revive=debug",
+            )
+            .stdout(Stdio::null())
+            .stderr(Stdio::piped())
+            .spawn()?;
+
+        // Give the node a moment to boot
+        Self::wait_ready(
+            &mut substrate_process,
+            Self::SUBSTRATE_READY_MARKER,
+            Duration::from_secs(10),
+        )?;
+
+        let mut proxy_process = Command::new(&self.eth_proxy_binary)
+            .arg("--dev")
+            .arg("--rpc-port")
+            .arg(proxy_rpc_port.to_string())
+            .arg("--node-rpc-url")
+            .arg(format!("ws://127.0.0.1:{substrate_rpc_port}"))
+            .env("RUST_LOG", "info,eth-rpc=debug")
+            .stdout(Stdio::null())
+            .stderr(Stdio::piped())
+            .spawn()?;
+
+        Self::wait_ready(
+            &mut proxy_process,
+            Self::ETH_PROXY_READY_MARKER,
+            Duration::from_secs(30),
+        )?;
+
+        self.process_substrate = Some(substrate_process);
+        self.process_proxy = Some(proxy_process);
+
+        Ok(())
+    }
+
+    fn generate_chainspec_file(&self, genesis_path: &str) -> anyhow::Result<PathBuf> {
+        let mut balances = self.extract_balance_from_genesis_file(genesis_path)?;
+
+        let mut chainspec_json: JsonValue =
+            serde_json::from_str(include_str!("default_chainspec.json"))?;
+
+        let existing_balances =
+            chainspec_json["genesis"]["runtimeGenesis"]["patch"]["balances"]["balances"]
+                .as_array()
+                .cloned()
+                .unwrap_or_default();
+
+        let mut merged_balances: Vec<(String, u128)> = existing_balances
+            .into_iter()
+            .filter_map(|val| {
+                if let Some(arr) = val.as_array() {
+                    if arr.len() == 2 {
+                        let account = arr[0].as_str()?.to_string();
+                        let balance = arr[1].as_f64()? as u128;
+                        return Some((account, balance));
+                    }
+                }
+                None
+            })
+            .collect();
+
+        merged_balances.append(&mut balances);
+
+        chainspec_json["genesis"]["runtimeGenesis"]["patch"]["balances"]["balances"] =
+            json!(merged_balances);
+
+        let temp_path =
+            std::env::temp_dir().join(format!("chainspec-{}.json", uuid::Uuid::new_v4()));
+
+        let mut temp_file = std::fs::File::create(&temp_path)?;
+        serde_json::to_writer_pretty(&mut temp_file, &chainspec_json)?;
+        temp_file.flush()?;
+
+        Ok(temp_path)
+    }
+
+    fn extract_balance_from_genesis_file(
+        &self,
+        genesis_path: &str,
+    ) -> anyhow::Result<Vec<(String, u128)>> {
+        let genesis_str = fs::read_to_string(genesis_path)?;
+        let genesis_json: Value = serde_json::from_str(&genesis_str)?;
```

Comment:
Same here, serializing into the Rust data-structure is more verbose but much nicer to work with.

Reply:
Yeah, please forget about what I suggested. Unfortunately it doesn't work out because of these dependencies are missing.